core/services/s4/postgres_orm.go

```diff
@@ -62,7 +62,7 @@ confirmed = EXCLUDED.confirmed,
 payload = EXCLUDED.payload,
 signature = EXCLUDED.signature,
 updated_at = NOW()
-WHERE t.version < EXCLUDED.version
+WHERE (t.version < EXCLUDED.version AND t.confirmed IS FALSE) OR (t.version <= EXCLUDED.version AND EXCLUDED.confirmed IS TRUE)
```

Comment:
Maybe add a clarifying comment that we never want to go back from CONFIRMED -> UNCONFIRMED on the same version.